File: solutions/haskell/space-age/1/src/SpaceAge.hs
module SpaceAge (Planet(..), ageOn) where

data Planet = Mercury
            | Venus
            | Earth
            | Mars
            | Jupiter
            | Saturn
            | Uranus
            | Neptune

earhSeconds :: Int
earhSeconds = 31557600

ageOn :: Planet -> Int -> Float
ageOn planet seconds =
  case planet of
    Earth   -> fromIntegral seconds / fromIntegral earhSeconds / 1.0
    Mercury -> fromIntegral seconds / fromIntegral earhSeconds / 0.2408467
    Venus   -> fromIntegral seconds / fromIntegral earhSeconds / 0.61519726
    Mars    -> fromIntegral seconds / fromIntegral earhSeconds / 1.8808158
    Jupiter -> fromIntegral seconds / fromIntegral earhSeconds / 11.862615
    Saturn  -> fromIntegral seconds / fromIntegral earhSeconds / 29.447498
    Uranus  -> fromIntegral seconds / fromIntegral earhSeconds / 84.016846
    Neptune -> fromIntegral seconds / fromIntegral earhSeconds / 164.79132

File: solutions/typescript/space-age/1/space-age.ts
const earthSeconds:number = 31557600;
const map: { [key:string]:number } = {
    mercury: 0.2408467,
    venus:   0.61519726,
    earth:   1.0,
    mars:    1.8808158,
    jupiter: 11.862615,
    saturn:  29.447498,
    uranus:  84.016846,
    neptune: 164.79132,
};

export function age(planet:string, seconds:number): number {
    return parseFloat((seconds / (map[planet] * earthSeconds)).toFixed(2));
}
